nestkernel/completed_checker.h

```diff
@@ -0,0 +1,112 @@
+/*
+ *  completed_checker.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef COMPLETED_CHECKER_H
+#define COMPLETED_CHECKER_H
+
+// C++ includes:
+#include <cassert>
+#include <cstddef>
+
+// Includes from nestkernel:
+#include "nest_types.h"
+#include "vp_manager.h"
+
+// Includes from sli:
+#include "dictdatum.h"
+
+namespace nest
+{
+
+/**
+ * A thread-safe array to coordinate progress across threads during
+ * gather operations. Uses an array of bools instead of an STL vector,
+ * since omp atomic operations can not be applied to vector
+ * assignments.
+ */
+
+class CompletedChecker
+{
+private:
+
+  /* array holding status values for all threads. must be of type
```

Comment:
Could you turn this and following member comments also into doxygen comments?

Reply:
sure!